src/main/java/org/spongepowered/common/mixin/core/entity/monster/MixinEntityCreeper.java

```diff
@@ -52,4 +76,98 @@
         this.ignite();
     }
 
+    /**
+     * We can't overwrite local variables any other way, unfortunately.
+     *
+     * @author kashike - 31/12/2015
+     */
+    @Overwrite
+    public void onUpdate() {
+        if (((EntityCreeper) (Object) this).isEntityAlive()) {
+            this.lastActiveTime = this.timeSinceIgnited;
+
+            if (((EntityCreeper) (Object) this).hasIgnited()) {
+                ((EntityCreeper) (Object) this).setCreeperState(1);
+            }
+
+            int i = ((EntityCreeper) (Object) this).getCreeperState();
+
+            if (i > 0 && this.timeSinceIgnited == 0) {
+                // Sponge start
+                List<Object> cause = Lists.newArrayList();
+
+                if (this.igniter != null) {
+                    cause.add(NamedCause.igniter(this.igniter));
+                }
+
+                if (this.stack != null) {
+                    cause.add(NamedCause.igniter(this.stack));
+                }
+
+                cause.add(NamedCause.source(this));
+
+                // Reset state
+                this.igniter = null;
+                this.stack = null;
+
+                ExplosionEvent.Pre event = SpongeEventFactory.createExplosionEventPre(Cause.of(cause), (World) this.worldObj);
+                SpongeImpl.postEvent(event);
+                if (event.isCancelled()) {
+                    i = 0;
+                } else {
+                    ((EntityCreeper) (Object) this).playSound("creeper.primed", 1.0F, 0.5F);
+                }
+                //((EntityCreeper) (Object) this).playSound("creeper.primed", 1.0F, 0.5F); // Sponge - move up into else
+                // Sponge end
+            }
+
+            this.timeSinceIgnited += i;
+
+            if (this.timeSinceIgnited < 0) {
+                this.timeSinceIgnited = 0;
+            }
+
+            if (this.timeSinceIgnited >= this.fuseTime) {
+                this.timeSinceIgnited = this.fuseTime;
+                this.explode();
+            }
+        }
+
+        super.entity$onUpdate();
```

Comment:
I've seen no other way to accomplish this after playing around for a good half hour, but if there is indeed another way I'll be happy to change it.